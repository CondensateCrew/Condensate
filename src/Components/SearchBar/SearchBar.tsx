import React, { useState } from 'react';

const SearchBar:React.FC = () => {
  const [ isSearching, setSearching ] = useState<boolean>(false);

  let bar;
  
  const toggleSearching = ():void => {
    setSearching(!isSearching)
  }

  !isSearching ? bar = (<img onClick={toggleSearching}>Arrow</img>)
  : bar = (<div><input type='text' placeholder='Search by topic'></input></div>)
  
  return (
    <>
    {bar}
    </>
  )
}
