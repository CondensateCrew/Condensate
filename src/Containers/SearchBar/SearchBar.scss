@import "../../css/variables.scss";
@import "../../css/index.scss";

.short-search-btn {
  background: $primary-gradient;
  background-position: center;
  background-repeat: no-repeat;
  background-size: $medium-text;
  box-shadow: $external-shadow;
  border-radius: $one-line-border-radius;
  display: -webkit-flex;
  display: -ms-flex;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: $one-line-padding-space;
  width: 4.25rem; height: 4.25rem;
  img {
    width: $big-text;
  }
}

.search-form {
  display: -webkit-flex;
  display: -ms-flex;
  display: flex;
  align-items: center;
  background: $primary-gradient;
  box-shadow: $external-shadow;
  border-radius: $one-line-border-radius;
  padding: $one-line-padding-space;
  width: 40%;
  input {
    @extend %regular-input;
    margin-top: 0;
    margin-right: $small-space;
  }
  img {
    width: $big-text;
  }
}
