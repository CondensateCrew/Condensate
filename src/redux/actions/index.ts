import {
  Action,
  Category,
  Brainstorm,
  UserLoginReceived,
  RandomWordCollection,
  Insight,
  ActionObject,
  IAddUserAction,
  ILogoutUserAction,
  IAddAllActionsAction,
  IAddNewActionAction,
  IAddAllBrainstormsAction,
  IAddNewBrainstormAction,
  IDeleteBrainstormAction,
  IAddAllCategoriesAction,
  IAddNewCategoryAction,
  IAddRandomWordCollectionsAction,
  IAddQuestionTemplatesAction,
  IAddSecretSauceAction,
  IAddCurrentBrainstormAction,
  IRemoveCurrentBrainstormAction,
  IAddChosenWordAction,
  IRemoveChosenWordAction,
  IAddInsightAction,
  IAddQueryAction,
  IRemoveQueryAction,
  IAddFilterAction,
  IRemoveFilterAction,
  IEndTimeAction,
  IReverseTimeAction,
  IEndInstructionsAction,
  IReverseInstructionsAction
} from 'interfaces';

// user reducer action creators
export const addUser = (user: UserLoginReceived): IAddUserAction => ({
  type: 'ADD_USER',
  user: user
});

export const logOutUser = (): ILogoutUserAction => ({
  type: 'LOG_OUT_USER',
});

// actions reducer action creators
export const addAllActions = (actions: Action[]): IAddAllActionsAction => ({
  type: 'ADD_ALL_ACTIONS',
  actions: actions
});

export const addNewAction = (action: Action): IAddNewActionAction => ({
  type: 'ADD_NEW_ACTION',
  action: action
});

// allBrainstorms reducer action creators
export const addAllBrainstorms = (brainstorms: Brainstorm[]): IAddAllBrainstormsAction => ({
  type: 'ADD_ALL_BRAINSTORMS',
  brainstorms: brainstorms
});

export const addNewBrainstorm = (brainstorm: Brainstorm): IAddNewBrainstormAction => ({
  type: 'ADD_BRAINSTORM',
  brainstorm: brainstorm
});

export const deleteBrainstorm = (id: number): IDeleteBrainstormAction => ({
  type: 'DELETE_BRAINSTORM',
  id: id
});

// categories reducer action creators
export const addAllCategories = (categories: Category[]): IAddAllCategoriesAction => ({
  type: 'ADD_ALL_CATEGORIES',
  categories: categories
});

export const addNewCategory = (category: Category): IAddNewCategoryAction => ({
  type: 'ADD_CATEGORY',
  category: category
});

// randomWordCollections reducer action creators
export const addRandomWordCollections = (collections: RandomWordCollection[]): IAddRandomWordCollectionsAction => ({
  type: 'ADD_ALL_COLLECTIONS',
  collections: collections
});

// questionTemplates reducer action creators
export const addQuestionTemplates = (templates: string[]): IAddQuestionTemplatesAction => ({
  type: 'ADD_ALL_TEMPLATES',
  templates: templates
});

// secretSauce reducer action creators
export const addSecretSauce = (secretSauce: string[]): IAddSecretSauceAction => ({
  type: 'ADD_SECRETE_SAUSE',
  secretSauce: secretSauce
});

// currentBrainstorm reducer action creators
export const addCurrentBrainstorm = (currentBrainstorm: Brainstorm): IAddCurrentBrainstormAction => ({
  type: 'ADD_CURRENT_BRAINSTORM',
  currentBrainstorm: currentBrainstorm
});

export const removeCurrentBrainstorm = (): IRemoveCurrentBrainstormAction => ({
  type: 'REMOVE_CURRENT_BRAINSTORM'
});

// chosenWords reducer action creators
export const addChosenWord = (chosenWord: string): IAddChosenWordAction => ({
  type: 'ADD_WORD',
  chosenWord: chosenWord
});

export const removeChosenWord = (chosenWord: string): IRemoveChosenWordAction => ({
  type: 'REMOVE_WORD',
  chosenWord: chosenWord
});

// insights reducer action creators
export const addInsight = (insight: Insight): IAddInsightAction => ({
  type: 'ADD_INSIGHT',
  insight: insight
});

// query reducer action creators
export const addQuery = (query: string): IAddQueryAction => ({
  type: 'ADD_QUERY',
  query: query
});

export const removeQuery = (): IRemoveQueryAction => ({
  type: 'REMOVE_QUERY'
});

// filter reducer action creators
export const addFilter = (filter: string): IAddFilterAction => ({
  type: 'ADD_FILTER',
  filter: filter
});

export const removeFilter = (): IRemoveFilterAction => ({
  type: 'REMOVE_FILTER'
});

// timeEnded reducer action creators
export const endTime = (): IEndTimeAction => ({
  type: 'END_TIME'
});

export const reverseTime = (): IReverseTimeAction => ({
  type: 'REVERSE_TIME'
});

// instructionsEnded reducer action creators
export const endInstructions = (): IEndInstructionsAction => ({
  type: 'END_INSTRUCTIONS'
});

export const reverseInstructions = (): IReverseInstructionsAction => ({
  type: 'REVERSE_INSTRUCTIONS'
});
