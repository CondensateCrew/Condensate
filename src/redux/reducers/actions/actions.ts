import { ActionObject } from 'interfaces';

type state = string[] | [ ];

const actions = (state: state = [ ], action: ActionObject) => {
  switch(action.type) {
    case 'ADD_ALL_ACTIONS':
      return action.actions;
    case 'ADD_NEW_ACTION':
      return [...state, action.action];
    case 'CLEAN_STORE':
      return [ ];
    default:
      return state;
  }
}

export default actions;
