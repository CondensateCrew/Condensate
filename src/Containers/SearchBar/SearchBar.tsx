import './SearchBar.scss';
import React, { useState, ChangeEvent, FormEvent, MouseEvent } from 'react';
import { useDispatch } from 'react-redux';
import { addQuery, removeQuery } from 'redux/actions';
import search from 'assets/search.svg';
import closeIcon from 'assets/close.svg';

const SearchBar: React.FC = () => {
  const dispatch = useDispatch();
  const [ isClicked, setIsClicked ] = useState<boolean>(false);
  const [ isSubmitted, setIsSubmitted ] = useState<boolean>(false);
  const [ query, setQuery ] = useState<string>('');

  const toggleForm = () => setIsClicked(!isClicked);

  const handleChanges = (event: ChangeEvent<HTMLInputElement>) => setQuery(event.target.value);

  const submitQuery = (event: FormEvent<HTMLFormElement> | MouseEvent): void => {
    event.preventDefault();
    if (query === '') return toggleForm();
    dispatch(addQuery(query));
    setIsSubmitted(!isSubmitted);
  };

  const cleanForm = () => {
    toggleForm();
    dispatch(removeQuery());
    setQuery('');
  };

  return (
    <>
      {(!isClicked)
        ? <button
          className="short-search-btn"
          onClick={toggleForm}>
            <img src={search} alt="search" onClick={submitQuery}/>
          </button>
        : <form onSubmit={submitQuery} className="search-form">
          <input
            type="text"
            placeholder="Search"
            value={query}
            onChange={handleChanges}/>
          {
            (!isSubmitted)
              ? <img src={search} alt="search" onClick={submitQuery}/>
              : <img src={closeIcon} alt="close" onClick={cleanForm}/>
          }
        </form>
      }
    </>
  )
}

export default SearchBar;
