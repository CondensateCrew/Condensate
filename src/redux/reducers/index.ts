import { combineReducers } from 'redux';
import user from '../reducers/user/user';
import allBrainstorms from '../reducers/allBrainstorms/allBrainstorms';
import categories from '../reducers/categories/categories';
import actions from '../reducers/actions/actions';
import randomWordCollections from '../reducers/randomWordCollections/randomWordCollections';
import questionTemplates from '../reducers/questionTemplates/questionTemplates';
import secretSauce from '../reducers/secretSauce/secretSauce';
import currentBrainstorm from '../reducers/currentBrainstorm/currentBrainstorm';
import chosenWords from '../reducers/chosenWords/chosenWords';
import insights from '../reducers/insights/insights';
import query from '../reducers/query/query';
import filter from '../reducers/filter/filter';
import timeEnded from '../reducers/timeEnded/timeEnded';
import instructionsEnded from '../reducers/instructionsEnded/instructionsEnded';

const rootReducer = combineReducers({
  user,
  allBrainstorms,
  categories,
  actions,
  randomWordCollections,
  questionTemplates,
  secretSauce,
  currentBrainstorm,
  chosenWords,
  insights,
  query,
  filter,
  timeEnded,
  instructionsEnded
});

export default rootReducer;
