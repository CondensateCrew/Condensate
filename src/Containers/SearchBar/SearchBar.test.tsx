import React from 'react';
import SearchBar from './SearchBar';
import { shallow } from 'enzyme';
import { addQuery, removeQuery } from 'redux/actions';

jest.mock('redux/actions')

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch
}));

describe('SearchBar component', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<SearchBar />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if btn is clicked', () => {
    wrapper.find('.short-search-btn').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if btn is clicked and query is submitted', () => {
    wrapper.find('.short-search-btn').simulate('click');
    wrapper.find('input').simulate('change', {target:{value: 'sth'}});
    wrapper.find('img[alt="search"]').simulate('click', {preventDefault: jest.fn()});
    expect(wrapper).toMatchSnapshot();
  });

  it('should call addQuery prop when submit is clicked', () => {
    wrapper.find('.short-search-btn').simulate('click');
    wrapper.find('input').simulate('change', {target:{value: 'sth'}});
    wrapper.find('img[alt="search"]').simulate('click', {preventDefault: jest.fn()});
    expect(addQuery).toHaveBeenCalledWith('sth');
  });

  it('should call removeQuery prop when submit is clicked', () => {
    wrapper.find('.short-search-btn').simulate('click');
    wrapper.find('input').simulate('change', {target:{value: 'sth'}});
    wrapper.find('img[alt="search"]').simulate('click', {preventDefault: jest.fn()});
    wrapper.find('img[alt="close"]').simulate('click');
    expect(removeQuery).toHaveBeenCalled();
  });
});
