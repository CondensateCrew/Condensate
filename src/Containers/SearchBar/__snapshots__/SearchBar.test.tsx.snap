// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component should match the snapshot 1`] = `
<Fragment>
  <button
    className="short-search-btn"
    onClick={[Function]}
  >
    <img
      alt="search"
      onClick={[Function]}
      src="search.svg"
    />
  </button>
</Fragment>
`;

exports[`SearchBar component should match the snapshot if btn is clicked 1`] = `
<Fragment>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
    <img
      alt="search"
      onClick={[Function]}
      src="search.svg"
    />
  </form>
</Fragment>
`;

exports[`SearchBar component should match the snapshot if btn is clicked and query is submitted 1`] = `
<Fragment>
  <form
    className="search-form"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value="sth"
    />
    <img
      alt="close"
      onClick={[Function]}
      src="close.svg"
    />
  </form>
</Fragment>
`;
